back/server/modules/battleshipRound.js

```diff
@@ -180,6 +186,17 @@ class BattleshipRound {
         return {x:x, y:y}
     }
 
+    targetNextShipTile(opMap) {
```

Comment:
this is an inefficient way of doing this. Your current implementation has us iterate through the grid **every single time** its the AI's turn, and checks tiles we know are hit already.

Instead, lets generate a list of "tiles to hit" at the start of the first AI turn that is stored in this class. Then, whenever its the AI's turn, grab the next item in the list, remove it from the list, and return that tile.